transmission/transmission.go

```diff
@@ -26,6 +26,7 @@ import (
 	"time"
 
 	"github.com/facebookgo/muster"
+	"github.com/valyala/gozstd"
```

Comment:
I don't think we can use this library in libhoney, since it relies on pre-compiled archive files which are only provided for a handful of platforms, and (as Emily experienced) don't link successfully on non-matching golang version. The DataDog library includes C source and so should build far more readily. Note I didn't choose DataDog for our internal use because it allocated a huge pile of memory when decoding, but that may not be true for encodes.

Reply:
In the worst case, we can wind up falling through to gzip if we can't zstd, no? or are you worried this won't compile at all on other platforms? anyways, it's sounding like this might be harder than I first thought, do you want to take over the microbenchmarking, @ianwilkes?

Reply:
Yeah, I can probably get this done later today.